{"priority": "major", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/osrf/sdformat.json"}, "html": {"href": "#!/osrf/sdformat"}, "avatar": {"href": "data/bytebucket.org/ravatar/{b6d52f9f-b070-41c0-807c-94af07ea375b}ts=1606789"}}, "type": "repository", "name": "sdformat", "full_name": "osrf/sdformat", "uuid": "{b6d52f9f-b070-41c0-807c-94af07ea375b}"}, "links": {"attachments": {"href": "data/repositories/osrf/sdformat/issues/97/attachments_page=1.json"}, "self": {"href": "data/repositories/osrf/sdformat/issues/97.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/sdformat/issues/97/watch"}, "comments": {"href": "data/repositories/osrf/sdformat/issues/97/comments_page=1.json"}, "html": {"href": "#!/osrf/sdformat/issues/97/windows-linking-error"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/sdformat/issues/97/vote"}}, "reporter": {"display_name": "Steve Peters", "uuid": "{2ccfed09-18b8-4921-8d58-15ef01092802}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D"}, "html": {"href": "https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1fb4816dad9e68337d3096f750951f6cd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsSP-1.png"}}, "nickname": "Steven Peters", "type": "user", "account_id": "557058:5de38267-b118-494c-aa76-4fab35448816"}, "title": "Windows linking error", "component": null, "votes": 0, "watches": 6, "content": {"raw": "Maybe it's a problem with my VM (Windows 8.1 on Parallels Desktop), but I just tried to rebuild sdformat on my windows VM, and it's giving boost linking errors. I configured it with `Debug` build type and it finds boost during configure ([configure log here](https://gist.github.com/scpeters/ecdd052a1a8501c8e32f)). However, it gives \"unresolved external\" linking errors like the following during build ([build log here](https://gist.github.com/scpeters/46eac3ed03e1c0f3c356)):\r\n\r\n~~~\r\nConverter.cc.obj : error LNK2019:\r\n unresolved external symbol \"__declspec(dllimport) public: __thiscall \r\n boost::filesystem::directory_entry::directory_entry(void)\" (__imp_??0directory_entry@filesystem@boost@@QAE@XZ)\r\n referenced in function \"public: __thiscall boost::filesystem::detail::dir_itr_imp::dir_itr_imp(void)\" (??0dir_itr_imp@detail@filesystem@boost@@QAE@XZ)\r\n~~~", "markup": "markdown", "html": "<p>Maybe it's a problem with my VM (Windows 8.1 on Parallels Desktop), but I just tried to rebuild sdformat on my windows VM, and it's giving boost linking errors. I configured it with <code>Debug</code> build type and it finds boost during configure (<a data-is-external-link=\"true\" href=\"https://gist.github.com/scpeters/ecdd052a1a8501c8e32f\" rel=\"nofollow\">configure log here</a>). However, it gives \"unresolved external\" linking errors like the following during build (<a data-is-external-link=\"true\" href=\"https://gist.github.com/scpeters/46eac3ed03e1c0f3c356\" rel=\"nofollow\">build log here</a>):</p>\n<div class=\"codehilite\"><pre><span></span>Converter.cc.obj : error LNK2019:\n unresolved external symbol &quot;__declspec(dllimport) public: __thiscall \n boost::filesystem::directory_entry::directory_entry(void)&quot; (__imp_??0directory_entry@filesystem@boost@@QAE@XZ)\n referenced in function &quot;public: __thiscall boost::filesystem::detail::dir_itr_imp::dir_itr_imp(void)&quot; (??0dir_itr_imp@detail@filesystem@boost@@QAE@XZ)\n</pre></div>", "type": "rendered"}, "assignee": null, "state": "resolved", "version": null, "edited_on": null, "created_on": "2015-06-08T23:15:41.172818+00:00", "milestone": null, "updated_on": "2017-03-01T15:09:48.861257+00:00", "type": "issue", "id": 97}